index.js

```diff
@@ -0,0 +1,30 @@
+const myNavv = document.getElementById('myNav');
+function openNav() {
+  myNavv.style.width = '100%';
+}
+function closeNav() {
+  document.getElementById('myNav').style.width = '0%';
+}
+const openNavv = document.getElementById('.myNav');
+const closeNavv = document.getElementById('.myNav');
+openNavv.addEventListener('click', openNav);
+closeNavv.addEventListener('click', closeNav);
+
+// data preserve
+const contactForm = document.querySelector('.upper-form-desk');
+const nameField = document.querySelectorAll('#Name');
+const emailAddressField = document.querySelector('#useremail');
+const messageField = document.querySelector('#msg');
+
+contactForm.addEventListener('input', () => {
+  const userData = {
+    name: nameField.value,
+    email: emailAddressField.value,
+    message: messageField.value,
+  };
+  localStorage.setItem('userData', JSON.stringify(userData));
+});
+const userDataFromLocalStorage = JSON.parse(localStorage.getItem('userData'));
+nameField.value = userDataFromLocalStorage.name;
+emailAddressField.value = userDataFromLocalStorage.email;
+messageField.value = userDataFromLocalStorage.message;
```

Comment:
Great job team🤝

- [ ] As stated in the project requirements, kindly ensure your project saves data to the local storage. I am not able to see data being saved to the local storage.
![local storage](https://github.com/solog0039/Portfolio-final/assets/63917725/a6eab46c-f913-4270-89e0-bef547ff562d)

- [ ] You may also ensure this project achieves the following just as stated in the project requirements.
![project requirements](https://github.com/solog0039/Portfolio-final/assets/63917725/b6aef7bb-ebad-4532-a96c-5ee51de810a1)

- [ ] Kindly also ensure when the user changes the content of any input field in the form the data object is updated and it is saved to the local storage.
- [ ] Kindly also ensure when the user loads the page, the code checks if there are any saved data in the local storage, and if so the input fields are pre-filled with the data.
![window onload](https://github.com/solog0039/Portfolio-final/assets/63917725/b2333444-9b80-4a59-849e-2258551fcc97)